File: Famicoco/app/src/main/java/com/example/mito/famicoco/TalkCustomAdapter.java
package com.example.mito.famicoco;

import android.content.Context;
import androidx.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;

import static com.example.mito.famicoco.MainActivity.myName;

class TalkCustomAdapter extends ArrayAdapter<CustomData> {
    private LayoutInflater layoutInflater;


    @Override
    public int getViewTypeCount() {
        return 2;
    }


    TalkCustomAdapter(Context c, int id, ArrayList<CustomData> list) {
        super(c, id, list);
        this.layoutInflater = (LayoutInflater) c.getSystemService(
                Context.LAYOUT_INFLATER_SERVICE);

    }

    @NonNull
    @Override
    public View getView(int position, View convertView, @NonNull ViewGroup parent) {
        View view;
        CustomData customData = getItem(position);
        if (!customData.getName().equals(myName)) {
            view = layoutInflater.inflate(R.layout.ie_row2, parent, false);
        } else {
            view = layoutInflater.inflate(R.layout.ie_row, parent, false);
        }
        ((ImageView) view.findViewById(R.id.ie2_icon))
                .setImageBitmap(customData.getIcon());
        ((TextView) view.findViewById(R.id.ie2_textView))
                .setText(customData.getName());
        ((TextView) view.findViewById(R.id.ie2_textView2))
                .setText(customData.getComment());
        ((TextView) view.findViewById(R.id.time_now))
                .setText(customData.getTime_now());
        return view;
    }

    //タッチ無効
    public boolean isEnabled(int position) {
        return false;
    }
}

File: Famicoco/app/src/main/java/com/example/mito/famicoco/TalkCustomAdapter.kt
package com.example.mito.famicoco

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView
import com.example.mito.famicoco.MainActivity.Companion.myName
import java.util.ArrayList

internal class TalkCustomAdapter(
        c: Context?,
        id: Int,
        list: ArrayList<CustomData>?
) : ArrayAdapter<CustomData>(
        c,
        id,
        list
) {
    private val layoutInflater: LayoutInflater = c?.getSystemService(
            Context.LAYOUT_INFLATER_SERVICE
    ) as LayoutInflater
    
    override fun getViewTypeCount(): Int {
        return 2
    }
    
    override fun getView(
            position: Int,
            convertView: View?,
            parent: ViewGroup
    ): View {
        val view: View
        val customData = getItem(position)
        view = if (customData?.name != myName) {
            layoutInflater.inflate(
                    R.layout.ie_row2,
                    parent,
                    false
            )
        } else {
            layoutInflater.inflate(
                    R.layout.ie_row,
                    parent,
                    false
            )
        }
        customData?.let {
            (view.findViewById<View>(R.id.ie2_icon) as ImageView)
                    .setImageBitmap(it.icon)
            (view.findViewById<View>(R.id.ie2_textView) as TextView).text = it.name
            (view.findViewById<View>(R.id.ie2_textView2) as TextView).text = it.comment
            (view.findViewById<View>(R.id.time_now) as TextView).text = it.timeNow
        }
        return view
    }
    
    // タッチ無効
    override fun isEnabled(position: Int): Boolean {
        return false
    }
}
